lesson_8/lesson-8-rusu-daniel/Tema_8.py

```diff
@@ -0,0 +1,74 @@
+class Vehicle(object):
+    
+    def __init__(self,model,transmission,weight,cubic_capacity,number_of_seats,consumtion):
+        self.transmission = transmission
+        self.weight = weight
+        self.cubic_capacity = cubic_capacity
+        self.number_of_seats = number_of_seats
+        self.consumtion = consumtion
+        self.model = model
+        
+    def is_vehicle(self, print_result=False):
+        if not print_result:
+            return self.number_of_seats > 0 and self.number_of_seats < 9 and self.weight > 1000 and self.cubic_capacity > 1.0 and self.transmission == 'Automatic' or self.transmission == 'Manual' or self.transmission =='Semi-Automatic'
```

Comment:
aici as face niste metode separate pentru fiecare variabila pe care o verifici si in metoda `is_vechicle` as chema metodele respective.

```
def check_seats(self):
    return self.number_of_seats > 0 and self.number_of_seats < 9

def check_weight(self):
    return self.weight > 1000

def is_vehicle(self, print_result=False):
    return all([self.check_seats(), self.check_weight()])
```
gasesti aici informatii despre all https://docs.python.org/3/library/functions.html#all